pytorch_lightning/plugins/training_type/ddp2.py

```diff
@@ -73,5 +73,6 @@ def _is_single_process_single_device(self) -> bool:
         return False
 
     def set_world_ranks(self):
-        self.cluster_environment.set_global_rank(self.node_rank)
-        self.cluster_environment.set_world_size(self.num_nodes)
+        if self.cluster_environment is not None:
+            self.cluster_environment.set_global_rank(self.node_rank)
+            self.cluster_environment.set_world_size(self.num_nodes)
```

Comment:
super nit: early returns for less branching
```suggestion
        if self.cluster_environment is None:
            return
        self.cluster_environment.set_global_rank(self.node_rank)
        self.cluster_environment.set_world_size(self.num_nodes)
```